bench/compare2.sh

```diff
@@ -42,37 +42,30 @@ run_v2() {
 run_checkout
 
 run_setup() {
-  if [ "$1" == "v1" ]; then
-    cd $tempdir/v1/
-    mojo build llama2.mojo
-  elif [ "$1" == "v2" ]; then
-    cd $tempdir/v2/
-    mojo build llama2.mojo
-  fi;
+  cd $tempdir/$1/
+  mojo build llama2.mojo
 }
 
 export -f run_v1
 export -f run_v2
 export -f run_setup
 
-run_setup v1
-# if exit code is not 0, exit
-if [ $? -ne 0 ]; then
-  echo "Error building v1"
-  exit 1
-fi
+IMPLEMENTATIONS=('v1' 'v2')
 
-run_setup v2
-# if exit code is not 0, exit
-if [ $? -ne 0 ]; then
-  echo "Error building v2"
-  exit 1
-fi
+# Loop through all versions in IMPLEMENTATIONS and set them up
+for ver in "${IMPLEMENTATIONS[@]}"; do
+  run_setup "$ver"
 
-IMPLEMENTATIONS='v1,v2'
-PARAM_MODELS='stories15M.bin,stories42M.bin,stories110M.bin'
+  if [ $? -ne 0 ]; then
+    echo "Error building $ver"
+    exit 1
+  fi
+done
 
-hypertune "run_{ver} {model}" \
+IMPLEMENTATIONS="v1,v2"
+PARAM_MODELS="stories15M.bin,stories42M.bin,stories110M.bin"
+
+hypertune --shell=bash "run_{ver} {model}" \
```

Comment:
this `--shell=bash` opens another session as I understood, so that's why it can't find `run_v1` 
I was executing it like this `bash compare2.sh` and I didn't set `--shell` for hypertune

Reply:
yes I was running the command with bash, I would run bash compare2.sh, and the same problem would appear. But my main shell is *zsh*, but I would also change my shell to *bash*, and then run bash compare2.sh, and I would get the same error, adding the --shell=bash command fix it for me, I try looking for other options, but I didn't find anything. I don't know a lot about shell scripting, but from what I understood is that maybe it tries to run the command in a parent shell session instead of sub-child session or I don't know, I don't know why hypertune (and I also tried with hyperfine) could't find the command.